### PR TITLE
Bedpetovcf rewrite

### LINKED ISSUE #74: bedpetovcf crashes if STRAND field not in INFO
Believe this is the same or similar to #2

### PR DESCRIPTION
**Pull request by ernfrid**
This is a rewrite of bedpetovcf. There is a new converter class that takes a Bedpe object and returns VCF Variant objects. As part of this rewrite, I have dropped the reconstruction of the INFO field of the secondary breakend from the INFO field of the primary breakend. It seemed unnecessary given that we store both if they are available and don't try to output both breakends if one is missing anyways.

I believe this has the following desirable properties:
* STRANDS is no longer required. (Fixes #74)
* The POS field of the secondary breakend no longer contains the POS of the primary breakend.
* The MATEID field for secondary breakends no longer (circularly) points to the entry itself.
* We no longer spend time fixing up the primary entries INFO field for certain fields.

It may have the following undesirable properties:
* May be slightly slower. I have not benchmarked.
* SECONDARY breakend lines now do not contain an END field (although it was frequently wrong before anyways)

I do not expect that I have uniformly fixed everything and would greatly appreciate a second pair of eyes on this.

### CODE REVIEW

File: svtools/bedpetovcfconverter.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        if o1 == '+':
+            return ('N', '')
+        else:
+            return ('', 'N')
+
+    def bnd_alt_string(self, o1, o2, chrom, pos):
+        '''
+        Given a Bedpe object generate the correct alt string for a BND VCF entry
+        '''
+        alt_string = '{3}{0}{1}:{2}{0}{4}'
```
Comment (cc2qe) [inline svtools/bedpetovcfconverter.py]: never seen this notation before, what does it do?

Comment (ernfrid) [inline svtools/bedpetovcfconverter.py]: `@staticmethod`?

Comment (cc2qe) [inline svtools/bedpetovcfconverter.py]: this is a staticmethod?
`alt_string = '{3}{0}{1}:{2}{0}{4}'`
it looks like just a string of placeholders

Comment (ernfrid) [inline svtools/bedpetovcfconverter.py]: Nope. That's a format string as discussed offline. I really liked this website: https://pyformat.info/

**Comment by cc2qe** [comment]
this looks great to me. clean and accurate

**Comment by ernfrid** [comment]
@cc2qe - Cool. Thanks for looking it over.

I'll leave this open in case anyone else wants to comment, but aim to merge at the end of the day.

File: svtools/bedpetovcfconverter.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        return alt_string.format(self.determine_sep(o2), chrom, pos, *self.determine_flanks(o1))
+
+    def convert(self, bedpe):
+        '''
+        Convert a bedpe object to Vcf object(s). Returns a list of entries.
+        '''
+        b1 = self.adjust_by_cipos(bedpe)
+        b2 = self.adjust_by_ciend(bedpe)
```
Comment (abelhj) [inline svtools/bedpetovcfconverter.py]: No big deal, but you could avoid adjust by ciend except in the BND case.

**Comment by ernfrid** [comment]
@abelhj - Good call. Fixed.

**Pull request merged**